### PR TITLE
Fixed Privilege-Related Issues

### PR DESCRIPTION
**Pull request by Hydrocharged**
Fixes the following issues:
* https://github.com/dolthub/dolt/issues/4081
* https://github.com/dolthub/dolt/issues/4082
* https://github.com/dolthub/dolt/issues/4085

### CODE REVIEW

File: enginetest/memory_engine_test.go [hunk 1]
```diff
@@ -524,6 +524,7 @@ func TestUserPrivileges(t *testing.T) {
 }
 
 func TestUserAuthentication(t *testing.T) {
+	queries.ServerAuthTests = queries.ServerAuthTests[:2]
```
Comment (zachmu) [inline enginetest/memory_engine_test.go:527]: ?

Comment (zachmu) [reply enginetest/memory_engine_test.go:527]: At least add a comment

Better yet, make a different var here with a descriptive name

**Review by zachmu — APPROVED** [review]
LGTM

File: enginetest/memory_engine_test.go (hunk 1 above)
Comment (Hydrocharged) [reply enginetest/memory_engine_test.go]: Leaving this in there was an accident lol, this was just to speed up testing by not running all of the tests at once

**Pull request merged**